LethalLevelLoader/Patches/TerminalManager.cs

```diff
@@ -81,6 +81,11 @@ internal static void CacheTerminalReferences()
 
         internal static bool LockedNodeEventTest(ref TerminalNode currentNode, ref TerminalNode loadNode)
         {
+            // This prevents an IndexOutOfRangeException when travelling to the company store
+            if (StartOfRound.Instance.levels.Length <= loadNode.displayPlanetInfo) {
```

Comment:
I also looked into this and my guess is that the exact reason is that for the "route company" TerminalNode displayPlanetInfo is set to its default value -1. 
Even if you look at the game code-base on Terminal::TextPostProcess you can see they are checking against displayPlanetInfo != -1

I patched the IL (look at the IL attached below) for the affected method in the latest DLL version of the mod and after testing it, seems to work:
```asm
    ldarg.1
    ldind.ref
    ldfld int32 ['Assembly-CSharp']TerminalNode::displayPlanetInfo
    ldc.i4.m1  // Push -1 onto the stack
    bne.un.s original  // Branch if not equal to -1
    ldc.i4.1  // Push true onto the stack
    ret  // Return true
    ···
    original:
    // Original code starts here
```
```suggestion
            if (loadNode.displayPlanetInfo == -1) {
```

Reply:
Good catch! The -1 case occurred to me later, but I didn't go back to update the PR. My group went with the suggested other mod that masks this problem as a stopgap workaround.